packages/controllers/src/permissions/Permission.ts

```diff
@@ -402,6 +426,59 @@ export type PermissionSpecificationConstraint = {
   validator?: PermissionValidatorConstraint;
 };
 
+/**
+ * The constraint for restricted method permission specification objects.
+ * Permissions that correspond to JSON-RPC methods are specified using objects
+ * that conform to this type.
+ *
+ * See the README for more details.
+ */
+export type RestrictedMethodSpecificationConstraint =
+  PermissionSpecificationBase<PermissionType.RestrictedMethod> & {
+    /**
+     * The implementation of the restricted method that the permission
+     * corresponds to.
+     */
+    methodImplementation: RestrictedMethod<any, any>;
+  };
+
+/**
+ * The constraint for endowment permission specification objects. Permissions
+ * that endow callers with some restricted resource are specified using objects
+ * that conform to this type.
+ *
+ * See the README for more details.
+ */
+export type EndowmentSpecificationConstraint =
+  PermissionSpecificationBase<PermissionType.Endowment> & {
+    /**
+     * Endowment permissions do not support caveats.
+     */
+    allowedCaveats: null;
```

Comment:
I guess this was easier than moving `allowedCaveats` from the base to `RestrictedMethodSpecificationConstraint`?

Reply:
Oh, I see the note about them having caveats in the future potentially.